core/cog-view.c

```diff
@@ -87,6 +91,9 @@ cog_view_get_property(GObject *object, unsigned prop_id, GValue *value, GParamSp
     case PROP_USE_KEY_BINDINGS:
         g_value_set_boolean(value, cog_view_get_use_key_bindings(self));
         break;
+    case PROP_VIEWPORT:
+        g_value_set_object(value, cog_view_get_viewport(self));
```

Comment:
We need to use `g_value_take_object()` here, because `cog_view_get_viewport()` adds a ref to the object, and then `g_value_set_object()` adds _another_ ref. The `_take_object()` function “takes ownership” of the object, that is: does not add a reference, and assumes it can unref the object when it does not need it anymore.

(Sorry that I didn't notice this in my previous review.)

Reply:
understood.